src/plot/series/mark-series.js

```diff
@@ -32,6 +33,44 @@ const predefinedClassName = 'rv-xy-plot__series rv-xy-plot__series--mark';
 const DEFAULT_STROKE_WIDTH = 1;
 
 class MarkSeries extends AbstractSeries {
+  shouldComponentUpdate(nextProps, nextState) {
+    const isMissingDataProp = !this.props.data || !nextProps.data;
+    const isMissingNullAccessorProp = !this.props.nullAccessor || !nextProps.nullAccessor;
+    return [
```

Comment:
There's no reason to recreate this array every time this func is called. Make a const array outside the class declaration with just the string prop names, and loop over that and pull values out of `this.props` and `nextProps` for each string name. More legible and a slight optimization.